zcnswap/swapfactory/bancor/service.go

```diff
@@ -0,0 +1,231 @@
+	ethutils "github.com/0chain/gosdk/zcnswap/utils"
+	cmn "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"go.uber.org/zap"
+	"math/big"
+	"time"
+)
+
+type ISwapService interface {
+	SwapWithConversionPath(pair *swapfactory.ExchangePair, from string) (*types.Transaction, error)
+	Approve(spender, fromToken, fromAddress cmn.Address, amountIn *big.Int) error
+	IncreaseAllowance(spender, fromToken, fromAddress cmn.Address, addedValue *big.Int) error
+	PackConvert(path *[]cmn.Address, amount, minReturn *big.Int, affiliate cmn.Address) ([]byte, error)
+	EstimateRate(from, to string, amount *big.Int) (*swapfactory.ExchangePair, error)
+}
+
+type swapService struct {
+	Client     *ethclient.Client
+	PrivateKey *ecdsa.PrivateKey
+}
+
+//func SwapWithConversionPath(path *[]cmn.Address, from string, amountIn, amountOut *big.Int, privateKey *ecdsa.PrivateKey, client *ethclient.Client) (*types.Transaction, error) {
+func (s *swapService) SwapWithConversionPath(pair *swapfactory.ExchangePair, from string) (*types.Transaction, error) {
+	var affiliateAccount = cmn.HexToAddress("0x0000000000000000000000000000000000000000")
+
+	fromAddress := cmn.HexToAddress(from)
+	spender := cmn.HexToAddress(config.Configuration.BancorAddress)
+	fromToken := cmn.HexToAddress(config.Configuration.UsdcTokenAddress)
+
+	bancorModule, err := contracts.NewIBancorNetwork(spender, s.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	balance, err := erc20.TokenBalance(fromAddress, fromToken, s.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	if balance.Cmp(pair.AmountIn) != 1 {
+		return nil, errors.New("500", "Not enough balance")
+	}
+
+	tokenInfo, err := erc20.TokenInfo(fromToken, s.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	// allowance
+	aCheckAllowanceResult, err := erc20.CheckAllowance(fromToken,
+		spender,
+		fromAddress,
+		pair.AmountIn,
+		tokenInfo.TokenSymbol == "ETH",
+		s.Client)
+	if err != nil {
+		l.Logger.Error("Allowance error", zap.Error(err))
+		return nil, err
+	}
+
+	convertPayload, err := s.PackConvert(pair.ConversionPath, pair.AmountIn, pair.AmountOut, affiliateAccount)
+
+	if !aCheckAllowanceResult.IsSatisfied {
+		err := s.Approve(spender, fromToken, fromAddress, pair.AmountIn)
+		if err != nil {
+			l.Logger.Error("Approve error", zap.Error(err))
+			return nil, err
+		}
+	}
+
+	value := big.NewInt(0)
+	opts, err := ethutils.NewSignedTransaction(convertPayload, from, spender.Hex(), value, s.PrivateKey, s.Client)
+	if err != nil {
+		l.Logger.Error("Signed transaction errorr", zap.Error(err))
+		return nil, err
+	}
+
+	// ropsten test network deployed old contract, hence using old method
+	//return bancorModule.ConvertByPath2(opts, *path, amountIn, amountOut, affiliateAccount)
+	return bancorModule.ConvertByPath(opts, *pair.ConversionPath, pair.AmountIn, pair.AmountOut, affiliateAccount, affiliateAccount, big.NewInt(0))
```

Comment:
move to config so we could change network for mainnet

Reply:
It depends on **eth client** network, which is initialized upper in the code.

`func Swap(swapAmount int64, tokenSource string) (string, error) {
	client, _ := zcncore.GetEthClient()`

https://github.com/0chain/gosdk/blob/91bf935b5282d4deb7882c05eaafb6ab48e719d4/zcncore/ethwallet.go#L30

This code already pickup network from config

Reply:
remove dead code then